compression/lz77.py

```diff
@@ -68,79 +75,80 @@ def compress(self, text: str) -> list:
 
             # find the next encoding phrase
             # - triplet with offset, length, indicator (the next encoding character)
-            (offset, length, indicator) = self._find_encoding_token(text, search_buffer)
+            token = self._find_encoding_token(text, search_buffer)
 
             # update the search buffer:
             # - add new characters from text into it
             # - check if size exceed the max search buffer size, if so, drop the
             #   oldest elements
-            search_buffer += text[: length + 1]
+            search_buffer += text[: token.length + 1]
             if len(search_buffer) > self.search_buffer_size:
                 search_buffer = search_buffer[-self.search_buffer_size :]
 
             # update the text
-            text = text[length + 1 :]
+            text = text[token.length + 1 :]
 
             # append the token to output
-            output.append((offset, length, indicator))
+            output.append(token)
 
         return output
 
-    def decompress(self, tokens: list) -> str:
-        """This method turns the list of tokens consisting of triplets of the form
+    def decompress(self, tokens: List[Token]) -> str:
+        """This method turns the List of tokens consisting of triplets of the form
         (offset, length, char), into an output string.
 
         Args:
-            tokens (list): Tokens (offset, length, char)
+            tokens (List[Token]): Tokens (offset, length, char)
 
         Returns:
             output (str): The decompressed text
 
         Tests:
             >>> lz77_compressor = LZ77Compressor(13, 6)
-            >>> lz77_compressor.decompress([(0, 0, 'c'), (0, 0, 'a'), (0, 0, 'b'), \
-                    (0, 0, 'r'), (3, 1, 'c'), (2, 1, 'd'), (7, 4, 'r'), (3, 5, 'd')])
+            >>> lz77_compressor.decompress([Token(0, 0, 'c'), Token(0, 0, 'a'), \
+                Token(0, 0, 'b'), Token(0, 0, 'r'), Token(3, 1, 'c'), \
+                Token(2, 1, 'd'), Token(7, 4, 'r'), Token(3, 5, 'd')])
             'cabracadabrarrarrad'
-            >>> lz77_compressor.decompress([(0, 0, 'a'), (0, 0, 'b'), (2, 2, 'c'), \
-                    (4, 3, 'a'), (2, 2, 'a')])
+            >>> lz77_compressor.decompress([Token(0, 0, 'a'), Token(0, 0, 'b'), \
```

Comment:
PEP8 (and CONTRIBUTING.md) says we should avoid backslash line termination in Python code because a space character to the right of the backslash will break the script on a change that is invisible to the reader.

Please try to use `# doctest: +NORMALIZE_WHITESPACE` or a custom `__str__()` method to avoid using backslash line termination.